presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixSplitManager.java

```diff
@@ -0,0 +1,252 @@
+            String inputQuery = phoenixClient.buildSql(
+                    connection,
+                    handle.getCatalogName(),
+                    handle.getSchemaName(),
+                    handle.getTableName(),
+                    layoutHandle.getDesiredColumns(),
+                    layoutHandle.getTupleDomain(),
+                    phoenixMetadata.getColumns(handle, false));
+
+            return new FixedSplitSource(getHadoopInputSplits(inputQuery).stream().map(split -> (PhoenixInputSplit) split).map(split -> {
+                List<HostAddress> addresses = getSplitAddresses(split);
+
+                return new PhoenixSplit(
+                        phoenixClient.getConnectorId(),
+                        handle.getCatalogName(),
+                        handle.getSchemaName(),
+                        handle.getTableName(),
+                        layoutHandle.getTupleDomain(),
+                        addresses,
+                        new WrappedPhoenixInputSplit(split));
+            }).collect(Collectors.toList()));
+        }
+        catch (IOException | InterruptedException | SQLException e) {
+            throw new PrestoException(PHOENIX_ERROR, e);
+        }
+    }
+
+    private List<HostAddress> getSplitAddresses(PhoenixInputSplit split)
+    {
+        List<HostAddress> addresses;
+        try {
+            String hostName = split.getLocations()[0];
+            HostAddress address = hostCache.get(hostName);
+            if (address == null) {
+                address = HostAddress.fromString(hostName);
+                hostCache.put(hostName, address);
+            }
+            addresses = ImmutableList.of(address);
+        }
+        catch (Exception e) {
+            log.warn("Failed to get split host addresses, will proceed without locality");
+            addresses = ImmutableList.of();
+        }
+        return addresses;
+    }
+
+    // use Phoenix MR framework to get splits
+    private List<InputSplit> getHadoopInputSplits(String inputQuery)
+            throws SQLException, IOException, InterruptedException
+    {
+        Configuration conf = new Configuration();
+        phoenixClient.setJobQueryConfig(inputQuery, conf);
+        Job job = Job.getInstance(conf);
+        return getSplits(job);
+    }
+
+    public List<InputSplit> getSplits(JobContext context)
+            throws IOException, InterruptedException
+    {
+        final Configuration configuration = context.getConfiguration();
+        final QueryPlan queryPlan = getQueryPlan(context, configuration);
+        final List<KeyRange> allSplits = queryPlan.getSplits();
+        return generateSplits(queryPlan, allSplits, configuration);
+    }
+
+    // mostly copied from PhoenixInputFormat, but without the region size calculations
+    private List<InputSplit> generateSplits(final QueryPlan qplan, final List<KeyRange> splits, Configuration config)
+            throws IOException
+    {
+        Preconditions.checkNotNull(qplan);
```

Comment:
```suggestion
        requireNonNull(queryPlan, "queryPlan is null");
```

(variable name, use `java.util.Objects#requireNonNull(T, java.lang.String)`)